tests/e2e/util/kubectl/kubectl.go

```diff
@@ -129,18 +129,31 @@ func (k Kubectl) ApplyString(yamlString string) error {
 
 // Apply applies the given yaml file to the cluster
 func (k Kubectl) Apply(yamlFile string) error {
-	err := k.ApplyWithLabels(yamlFile, "")
-	return err
+	return k.applyWithOptions("-f " + yamlFile)
 }
 
 // ApplyWithLabels applies the given yaml file to the cluster with the given labels
 func (k Kubectl) ApplyWithLabels(yamlFile, label string) error {
-	cmd := k.build(" apply " + labelFlag(label) + " -f " + yamlFile)
+	return k.applyWithOptions(labelFlag(label) + " -f " + yamlFile)
+}
+
+// ApplyKustomize applies the given kustomization file to the cluster
+func (k Kubectl) ApplyKustomize(kustomizationDirectory string) error {
+	return k.applyWithOptions("-k " + kustomizationDirectory)
+}
+
+// ApplyKustomizeWithLabels applies the given kustomization file to the cluster with the given labels
+func (k Kubectl) ApplyKustomizeWithLabels(kustomizationDirectory, label string) error {
+	return k.applyWithOptions(labelFlag(label) + " -k " + kustomizationDirectory)
+}
+
+// applyWithOptions is a helper function to apply resources with specific options given as a string
+func (k Kubectl) applyWithOptions(options string) error {
+	cmd := k.build(" apply " + options)
```

Comment:
```suggestion
func (k Kubectl) applyWithOptions(options ...string) error {
	cmd := []string{"apply"}
	cmd = append(cmd, options...)
	_, err := k.executeCommand(k.build(strings.Join(cmd, " ")))
```

It makes sense to use the ellipsis here as well.
Then callers can call more naturally and not worry about space between the options :)